### ISSUE TITLE
Swaylock crashes

### ISSUE THREAD

**Issue by cedrichaase** [swaywm/sway#1506]
Swaylock crashes whenever I type in just over 40 random characters into the "password prompt".

I pasted the stacktrace and output of `sway -d` here:
https://gist.github.com/cedrichaase/fd3edfa8cd728630ac5adc7b34a2278f

I'm running sway/swaylock version 0.15.0.
If there's any more information I can provide, please let me know.

**Comment by cedrichaase** [comment]
Push.
I might have failed to convey this, but this is a blatant security issue. If anybody can unlock my screen by typing in random stuff for a while, it makes swaylock rather pointless.

As indicated by the stack trace, the error seems to occur during the process of rendering the "typing indicator". I've toyed around with the code a little, but seeing as I have very limited experience with C and no experience with pixman/cairo, the best I could come up with was disabling rendering of said indicator altogether. This stops the error from occuring, but yeah, disabling a feature is not the same as fixing it :)

So, can anybody reproduce this?

**Comment by SirCmpwn** [comment]
Try the latest 0.15 branch.

**Comment by cedrichaase** [comment]
I had to keep hammering away at my keyboard a little longer this time, but swaylock still crashes for me at `c08c805` (latest commit in 0.15 at the time of writing). I added the stack trace for this version to the gist mentioned in the original post. It does look slightly different than before, although the error still seems to occur in `cairo_paint`.

**Comment by martinetd** [comment]
Sounds like a duplicate of #918, do you have at least two with one scaled and one not?
Curious if that would happen with wlroots as well, should be able to test pretty soon...

**Comment by cedrichaase** [comment]
Yeah, sounds like the same issue all right. Sorry, guess I overlooked that.
Indeed, I'm using a 13" MacBook Pro Retina (scaled) with a 1080p (non-scaled) display configured:
```
output eDP-1    scale 2     pos 320 1080
output DP-1     scale 1     pos 0   0
output DP-2     scale 1     pos 0   0
```
Though, while testing, the external (non-scaled) monitor was not connected.

**Comment by SirCmpwn** [comment]
Your stack traces are not terribly useful. Compile sway and wlc with debug symbols enabled.

**Comment by cedrichaase** [comment]
Yeah, I thought I might have to do that. Unfortunately, I have no idea how. Can you explain or point me to some documentation?
